website/content/docs/other-specifications/volume/host.mdx

```diff
@@ -101,12 +101,50 @@ the API.
 
 - `plugin_id` `(string)` - The ID of the [dynamic host volume
   plugin][dhv_plugin] that manages this volume. Required for volume
-  creation. Nomad has one built-in plugin called `mkdir`, which creates a
-  directory on the host in the Nomad agent's [host_volumes_dir][].
+  creation. Nomad has one built-in plugin called [`mkdir`][mkdir_plugin].
 
 - `type` `(string: <required>)` - The type of volume. Must be `"host"` for
   dynamic host volumes.
 
+## mkdir plugin
+
+Nomad has one built-in plugin called `mkdir`, which creates a directory on the
+host in the Nomad agent's [host_volumes_dir][]. The directory name is the
+volume's ID.
+
+`mkdir` ignores `capacity_min` and `capacity_max` volume configuration,
+since it has no way of enforcing them.
+
+### mkdir parameters
+
+- `mode` `(string: <optional>)` - [Numeric
+notation](https://en.wikipedia.org/wiki/File-system_permissions#Numeric_notation)
+to apply to the created directory. Defaults to "0700", so only the owner can
+access it.
+- `uid` `(string: <optional>)` - User ID to own the directory. Defaults to the
+user running the Nomad client agent, which is commonly root ("0").
+- `gid` `(string: <optional>)` - Group ID to own the directory. Defaults to the
+user running the Nomad client agent, which is commonly root ("0").
```

Comment:
Even though it's obvious to me in the way you've written this, I would still explicitly point out that the ID is an ID and not a name to emphasize that's the case even though the parameter is a string type.

Reply:
I _could_ say that it's an int here, because the volume spec accepts that. it just becomes a string later for me in code, but that's not something the user really needs to think about.

Reply:
Oh! Are we enjoying automatic conversion from HCL2 here?

Reply:
mmm, good guess, but nope. you might laugh.

* we're using hcl 1 here: https://github.com/hashicorp/nomad/blob/cc99e8f/command/volume_register.go#L138
   * at this point, it's a string
* we decode it to a `map[string]any` here: https://github.com/hashicorp/nomad/blob/cc99e8f/command/volume_create_host.go#L220-L221`
   * still a string
* we turn it into a `HostVolume` here: https://github.com/hashicorp/nomad/blob/cc99e8f/command/volume_create_host.go#L235
   * our friend _mapstructure_ makes a cameo: it looks at the type of [`HostVolume.Parameters`](https://github.com/hashicorp/nomad/blob/main/api/host_volumes.go#L59), and turns all the map vals into strings.